examples/agui/server/externaltool/agentnode_llmagent/client.py

```
@@ -0,0 +1,167 @@
+    final_text = collect_final_run(stream_events(args.endpoint, payload, args.timeout))
+    print("\nFinal answer:")
+    print(final_text or "(no final text)")
+
+
+def stream_events(endpoint: str, payload: dict, timeout: float):
+    request = urllib.request.Request(
+        endpoint,
+        data=json.dumps(payload).encode("utf-8"),
+        headers={"Content-Type": "application/json"},
+        method="POST",
+    )
```

Ruff (0.15.15)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)